javascript/src/api.ts

```diff
@@ -41,6 +41,43 @@ export type MapChanges = Map<
   }
 >;
 
+/**
+ * A factory interface for creating `ISharedDocument` objects.
+ */
+export interface IFactory {
+  /**
+   * Create a new `ISharedDocument` instance.
+   */
+  createNew(options: IFactory.IOptions): ISharedDocument;
+}
+
+/**
+ * The namespace for `IFactory`.
+ */
+export namespace IFactory {
+  /**
+   * The options used to instantiate a ISharedDocument
+   */
+  export interface IOptions {
+    /**
+     * The path of the file.
+     */
+    path: string;
+    /**
+     * The format of the document.
+     */
+    format: string;
+    /**
+     * The content type of the document.
+     */
+    contentType: string;
+    /**
+     * Wether the document is collaborative or not.
+     */
```

Comment:
Let's define the default value here.

```suggestion
     * 
     *  The default value is `true`.
     */
```